mapper.cpp

```diff
@@ -474,7 +532,31 @@ vector<Alignment> Mapper::align_threaded(Alignment& alignment, int& kmer_count,
         // Grab the map from node ID to kmer start positions for this particular kmer.
         auto& kmer_positions = positions.at(i);
         // Fill it in, since we know there won't be too many to work with.
-        index->get_kmer_positions(k, kmer_positions);
+        
+        if(gcsa) {
+            // We need to fill in this vector with the GCSA nodes and then convert.
+            std::vector<gcsa::node_type> gcsa_nodes;
+            gcsa->locate(gcsa_range, gcsa_nodes);
+            
+            for(gcsa::node_type gcsa_node : gcsa_nodes) {
+                if(gcsa::Node::rc(gcsa_node)) {
+                    // We found a kmer on the reverse strand. The old index
+                    // didn't handle these, so we ignore them. TODO: figure out
+                    // how to account for them.
+                    continue;
+                }
+                // Decode the result's ID and offset and record it
+                kmer_positions[gcsa::Node::id(gcsa_node)].push_back(gcsa::Node::offset(gcsa_node));
+            }
+            
+        } else if(index) {
+           index->get_kmer_positions(k, kmer_positions);
+        } else {
+            cerr << "error:[vg::Mapper] no search index present" << endl;
+            exit(1);
+        }
+        
+        
         // ignore this kmer if it has too many hits
         // typically this will be filtered out by the approximate matches filter
         if (kmer_positions.size() > hit_max) kmer_positions.clear();
```

Comment:
This could be moved earlier in the GCSA case, as we get the precise estimate up-front.